### PR TITLE
Have separate-architectures flag include hip solutions

### PR DESCRIPTION
**Pull request by JeremyAdamHart**
rocBLAS was seeing test failures when trying to use the separate-architectures flag from PR #1477 due to hip solutions not being included in the architecture specific TensileLibrary files. This PR makes two changes to fix this:
1. Adds hip solutions to each architecture file
2. Remove reindexing (which ensured each solution in a TensileLibrary file had a distinct index from solutions in other TensileLibrary files) The first change results in duplication of the hip solutions for each .dat file. There are a total of 707 hip kernels per architecture, compared to 40,000 kernels in total. The second change was originally intended for easy dynamic c++ library merging at runtime, however that code was cut from PR #1477, and at this point it introduces more risk of error, while not helping any current functionality. I think it would be worthwhile for me to revisit these changes after the release to create a separate TensileLibrary_hip.dat file with all of the hip solutions in it, and then reintroduce the reindexing and dynamic merging code when we have more time to confirm it's working correctly.

### CODE REVIEW

**Comment by zaliu** [comment]
Since testing won't finish before Friday 3/18, we need @bragadeesh to approve merging - assuming it passes CI overnight.  Thanks.

**Comment by zaliu** [comment]
rocBLAS [#1117](https://github.com/ROCmSoftwarePlatform/rocBLAS-internal/pull/1117) with this PR passed rocblas-test fully on gfx90a.  gfx906 testing is in progress.

**Review by bragadeesh — APPROVED** [review]
Approving since this one is important
Let's make sure rocblas extended tests are good in all arch

@JeremyAdamHart @rkamd @zaliu

**Review by zaliu — APPROVED** [review]
Given the very tight time constraint, please squash-merge as soon as @benjaminulmer reviewed and OKed this PR so that [#1117](https://github.com/ROCmSoftwarePlatform/rocBLAS-internal/pull/1117) testing can begin.  Thanks.

**Review by benjaminulmer — APPROVED** [review]
lgtm. The gfx906 failure isn't related to these changes

**Pull request merged**
